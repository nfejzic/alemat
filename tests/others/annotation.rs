use alemat::{
    children,
    elements::{Annotation, Ident, Operator},
    MathMl,
};

#[test]
fn annotation() {
    let out = MathMl::with_content(
        Annotation::builder()
            .content(String::from("This is an annotation"))
            .build(),
    )
    .render();

    crate::snap_test!(out, name: "others_annotation");
}

#[test]
fn annotation_xml() {
    let out = MathMl::with_content(
        Annotation::builder()
            .content(children![
                Ident::from("x"),
                Operator::plus(),
                Ident::from("y"),
            ])
            .build(),
    )
    .render();

    crate::snap_test!(out, name: "others_annotation_xml");
}
