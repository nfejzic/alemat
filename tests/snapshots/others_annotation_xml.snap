---
source: tests/others/annotation.rs
expression: out
---
<math><annotation-xml><mi>x</mi><mo>+</mo><mi>y</mi></annotation-xml></math>
